# Initial Data Exploration
This notebook is available to get familiar with the dataset and with using Jupyter notebooks.

**Other useful links:**
- <a href="https://github.com/schmitzmelissa/DSCT-Capstone01">Full Github Repository</a>
- <a href="https://www.kaggle.com/footprintnetwork/ecological-footprint">Kaggle Dataset</a>
- <a href="https://www.footprintnetwork.org/">Data Source Website</a>
- <a href="https://www.footprintnetwork.org/resources/glossary/">Glossary of Relevant Terms</a>

## Dataset Info
(acquired through the below code)

## Proposed Next Steps
- Convert region names to shorter abbreviations

In [2]:
import pandas as pd

# Allows better display of DataFrames
from IPython.display import display

# Create DataFrame from CSV file
countries = pd.read_csv('countries.csv')

# Confirm it is a DataFrame
type(countries)

pandas.core.frame.DataFrame

In [3]:
# Determine how many countries are in the list
len(countries)

188

**Discoveries:**
- This dataset is from 2016 and contains 188 countries total.
- As of 2018, there are 196-241 countries and territories (depends on political questions).
- 5 countries with name changes since 2016 are either recorded with the old name or not recorded (~2%)

**Conclusion:** Most countries of the world have been recognized with this data.

In [4]:
# Get a look at the composition of the DataFrame
countries.head(5)

Country                    Region  Population (millions)   HDI  \
0          Afghanistan  Middle East/Central Asia                  29.82  0.46   
1              Albania   Northern/Eastern Europe                   3.16  0.73   
2              Algeria                    Africa                  38.48  0.73   
3               Angola                    Africa                  20.82  0.52   
4  Antigua and Barbuda             Latin America                   0.09  0.78   

  GDP per Capita  Cropland Footprint  Grazing Footprint  Forest Footprint  \
0        $614.66                0.30               0.20              0.08   
1      $4,534.37                0.78               0.22              0.25   
2      $5,430.57                0.60               0.16              0.17   
3      $4,665.91                0.33               0.15              0.12   
4     $13,205.10                 NaN                NaN               NaN   

   Carbon Footprint  Fish Footprint      ...       Cropland  Grazing Land  \
0              0.18            0.00      ...           0.24          0.20   
1              0.87            0.02      ...           0.55          0.21   
2              1.14            0.01      ...           0.24          0.27   
3              0.20            0.09      ...           0.20          1.42   
4               NaN             NaN      ...            NaN           NaN   

   Forest Land  Fishing Water  Urban Land  Total Biocapacity  \
0         0.02           0.00        0.04               0.50   
1         0.29           0.07        0.06               1.18   
2         0.03           0.01        0.03               0.59   
3         0.64           0.26        0.04               2.55   
4          NaN            NaN         NaN               0.94   

   Biocapacity Deficit or Reserve  Earths Required  Countries Required  \
0                           -0.30             0.46                1.60   
1                           -1.03             1.27                1.87   
2                           -1.53             1.22                3.61   
3                            1.61             0.54                0.37   
4                           -4.44             3.11                5.70   

   Data Quality  
0             6  
1             6  
2             5  
3             6  
4             2  

[5 rows x 21 columns]

In [5]:
# Too many columns to print neatly in one shot, so splitting into chunks for better visibility
country_IDs = countries.iloc[:,:5]
country_footprints = countries.iloc[:,5:10]
country_resources = countries.iloc[:,10:15]
country_totals = countries.iloc[:,15:]

In [6]:
# Print out heads to ensure correct slicing
country_IDs.head(10)

# print(country_footprints.head(10))
# print(country_resources.head(10))
# print(country_totals.head(10))

Country                    Region  Population (millions)   HDI  \
0          Afghanistan  Middle East/Central Asia                  29.82  0.46   
1              Albania   Northern/Eastern Europe                   3.16  0.73   
2              Algeria                    Africa                  38.48  0.73   
3               Angola                    Africa                  20.82  0.52   
4  Antigua and Barbuda             Latin America                   0.09  0.78   
5            Argentina             Latin America                  41.09  0.83   
6              Armenia  Middle East/Central Asia                   2.97  0.73   
7                Aruba             Latin America                   0.10   NaN   
8            Australia              Asia-Pacific                  23.05  0.93   
9              Austria            European Union                   8.46  0.88   

  GDP per Capita  
0        $614.66  
1      $4,534.37  
2      $5,430.57  
3      $4,665.91  
4     $13,205.10  
5     $13,540.00  
6      $3,426.39  
7            NaN  
8     $66,604.20  
9     $51,274.10

In [7]:
# Rename columns for better readability
country_IDs = country_IDs.rename(columns = {'Population (millions)':'Pop (M)','GDP per Capita':'GDP per cap'})
country_IDs.head(10)

Country                    Region  Pop (M)   HDI GDP per cap
0          Afghanistan  Middle East/Central Asia    29.82  0.46     $614.66
1              Albania   Northern/Eastern Europe     3.16  0.73   $4,534.37
2              Algeria                    Africa    38.48  0.73   $5,430.57
3               Angola                    Africa    20.82  0.52   $4,665.91
4  Antigua and Barbuda             Latin America     0.09  0.78  $13,205.10
5            Argentina             Latin America    41.09  0.83  $13,540.00
6              Armenia  Middle East/Central Asia     2.97  0.73   $3,426.39
7                Aruba             Latin America     0.10   NaN         NaN
8            Australia              Asia-Pacific    23.05  0.93  $66,604.20
9              Austria            European Union     8.46  0.88  $51,274.10

In [8]:
# Look for columns with NaN values
print(len(countries.loc[:, countries.isnull().any()]))

188


In [19]:
print('Shape:',countries.shape,'\n\n')
print('Columns:',countries.columns,'\n\n')
print('Info:\n')
countries.info()

Shape: (188, 21) 


Columns: Index(['Country', 'Region', 'Population (millions)', 'HDI', 'GDP per Capita',
       'Cropland Footprint', 'Grazing Footprint', 'Forest Footprint',
       'Carbon Footprint', 'Fish Footprint', 'Total Ecological Footprint',
       'Cropland', 'Grazing Land', 'Forest Land', 'Fishing Water',
       'Urban Land', 'Total Biocapacity', 'Biocapacity Deficit or Reserve',
       'Earths Required', 'Countries Required', 'Data Quality'],
      dtype='object') 


Info:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 188 entries, 0 to 187
Data columns (total 21 columns):
Country                           188 non-null object
Region                            188 non-null object
Population (millions)             188 non-null float64
HDI                               172 non-null float64
GDP per Capita                    173 non-null object
Cropland Footprint                173 non-null float64
Grazing Footprint                 173 non-null float64
Forest Footprint     

- The data has 188 rows and 21 columns.
- The column names as-is are fairly consistent, but not all of them explicitly include units (units were inferred in the proposal from other data on the source website).
- Apparently 188 columns have NaN values...
 - So every single row contains at least one NaN?
 - (Seems weird to me... Also, I'm pretty sure there's AT LEAST 1 country that has all the columns filled out...)
 - I think I'm just using the wrong command.
 
 
 **Don't forget:** I split up the columns into separate DataFrames... So analysis in above two cells was on original DataFrame.

In [21]:
countries[countries.HDI.isnull()]

Country         Region  \
7                                     Aruba  Latin America   
18                                  Bermuda  North America   
24                   British Virgin Islands  Latin America   
33                           Cayman Islands  Latin America   
43                            Côte d'Ivoire         Africa   
62                            French Guiana  Latin America   
63                         French Polynesia   Asia-Pacific   
71                               Guadeloupe  Latin America   
91   Korea, Democratic People's Republic of   Asia-Pacific   
108                              Martinique  Latin America   
115                              Montserrat  Latin America   
120                                   Nauru   Asia-Pacific   
123                           New Caledonia   Asia-Pacific   
139                                 Réunion         Africa   
156                                 Somalia         Africa   
184               Wallis and Futuna Islands   Asia-Pacific   

     Population (millions)  HDI GDP per Capita  Cropland Footprint  \
7                     0.10  NaN            NaN                 NaN   
18                    0.06  NaN     $70,626.30                 NaN   
24                    0.03  NaN            NaN                 NaN   
33                    0.06  NaN            NaN                0.36   
43                   19.84  NaN      $1,016.83                0.51   
62                    0.24  NaN            NaN                0.07   
63                    0.27  NaN            NaN                0.75   
71                    0.46  NaN            NaN                0.11   
91                   24.76  NaN            NaN                0.28   
108                   0.40  NaN            NaN                0.13   
115                   0.00  NaN            NaN                 NaN   
120                   0.01  NaN            NaN                 NaN   
123                   0.25  NaN            NaN                0.68   
139                   0.86  NaN            NaN                0.14   
156                  10.20  NaN            NaN                0.16   
184                   0.01  NaN            NaN                 NaN   

     Grazing Footprint  Forest Footprint  Carbon Footprint  Fish Footprint  \
7                  NaN               NaN               NaN             NaN   
18                 NaN               NaN               NaN             NaN   
24                 NaN               NaN               NaN             NaN   
33                0.45              1.23              3.56            0.05   
43                0.06              0.22              0.26            0.14   
62                0.06              0.46              1.58            0.17   
63                0.65              0.12              2.39            0.82   
71                0.03              0.16              2.61            0.31   
91                0.00              0.14              0.68            0.02   
108               0.02              0.12              1.73            0.04   
115                NaN               NaN               NaN             NaN   
120                NaN               NaN               NaN             NaN   
123               0.62              0.18              1.78            0.32   
139               0.01              0.15              3.27            0.09   
156               0.43              0.52              0.07            0.01   
184                NaN               NaN               NaN             NaN   

         ...       Cropland  Grazing Land  Forest Land  Fishing Water  \
7        ...            NaN           NaN          NaN            NaN   
18       ...            NaN           NaN          NaN            NaN   
24       ...            NaN           NaN          NaN            NaN   
33       ...           0.00          0.02         0.19           0.11   
43       ...           0.88          0.30         0.48           0.04   
62       ...           0.

In [24]:
# Noticed that there seem to be units on the Data Quality
countries['Data Quality'].value_counts()

5     66
6     60
3B    29
3L    18
3T     7
2      6
4      2
Name: Data Quality, dtype: int64

Data Quality score information can be found <a href="https://www.footprintnetwork.org/data-quality-scores/">here</a>.

The purpose of these scores is to rate the quality of data that has been produced and maintained from various data sources since 1961. This may include missing, unavailable, incomplete, or error-containing data. The quality score, therefore, is a measure of confidence in DQ.

- 1-3 is the time series score
- A-D is the latest year score

(not entirely sure what that means... needs to be researched furter)

**Summary of DQ values in dataset:**
- **3B:** No component of BC or EF is unreliable or unlikely for the latest data year. Some individual components of the EF or BC are unlikely in the latest data year. The total EF and BC time series results are not significantly affected by unlikely data.
- **3L:** ???
- **3T:** ???

In [25]:
# Another starting place to look for values that don't fit
countries.describe()

Population (millions)        HDI  Cropland Footprint  \
count             188.000000  172.00000          173.000000   
mean               37.342372    0.68636            0.578208   
std               140.756836    0.15604            0.355691   
min                 0.000000    0.34000            0.070000   
25%                 2.037500    0.55750            0.350000   
50%                 7.970000    0.72000            0.520000   
75%                24.870000    0.80250            0.700000   
max              1408.040000    0.94000            2.680000   

       Grazing Footprint  Forest Footprint  Carbon Footprint  Fish Footprint  \
count         173.000000        173.000000        173.000000      173.000000   
mean            0.263179          0.373815          1.804913        0.122486   
std             0.352067          0.359349          1.898283        0.158427   
min             0.000000          0.010000          0.000000        0.000000   
25%             0.080000          0.170000          0.420000        0.020000   
50%             0.180000          0.260000          1.140000        0.070000   
75%             0.320000          0.460000          2.600000        0.150000   
max             3.470000          3.030000         12.650000        0.820000   

       Total Ecological Footprint    Cropland  Grazing Land  Forest Land  \
count                  188.000000  173.000000    173.000000   173.000000   
mean                     3.317606    0.531850      0.456590     2.459191   
std                      2.370931    0.672567      1.014738    10.593956   
min                      0.420000    0.000000      0.000000     0.000000   
25%                      1.482500    0.180000      0.030000     0.060000   
50%                      2.740000    0.350000      0.120000     0.340000   
75%                      4.640000    0.590000      0.340000     1.170000   
max                     15.820000    5.420000      8.230000    95.160000   

       Fishing Water  Urban Land  Total Biocapacity  \
count     173.000000  173.000000         188.000000   
mean        0.595145    0.067110           4.019681   
std         1.661872    0.054844          11.689075   
min         0.000000    0.000000           0.050000   
25%         0.030000    0.030000           0.675000   
50%         0.110000    0.050000           1.310000   
75%         0.370000    0.090000           2.815000   
max        16.070000    0.270000         111.350000   

       Biocapacity Deficit or Reserve  Earths Required  Countries Required  
count                      188.000000       188.000000          188.000000  
mean                         0.702074         1.915745            4.037397  
std                         11.771339         1.369624           12.444616  
min                        -14.140000         0.240000            0.020000  
25%                         -1.935000         0.855000            0.942500  
50%                         -0.730000         1.580000            1.705000  
75%                          0.212500         2.677500            2.847500  
max                        109.010000         9.140000          159.470000

- Max much higher than the mean (indicates a possible outlier)
 - Population
 - Cropland
 - Grazing Footprint
 - Forest land
 
Max and mean look evenly distributed:
- HDI
- 
    
**Thought process:** *Should I include it? Is it a true observation? Should I transform it?*